src/providers/pkcs11_provider/mod.rs

```diff
@@ -41,6 +43,12 @@ const SUPPORTED_OPCODES: [Opcode; 8] = [
     Opcode::PsaAsymmetricEncrypt,
 ];
 
+#[derive(Debug)]
+struct KeyMetadata {
```

Comment:
Could probably be moved in `key_metadata.rs`?

Reply:
Well if there's already a cache in the key info manager, there's no reason to have this structure - I only created it to add the public key ID in the other commit!